prow/jira/jira.go

```diff
@@ -214,11 +245,47 @@ func (jc *client) JiraURL() string {
 	return jc.url
 }
 
+// Used determines whether the client has been used
+func (jc *client) Used() bool {
+	return jc.transport.used
+}
+
 type bearerAuthRoundtripper struct {
 	generator BearerAuthGenerator
 	upstream  http.RoundTripper
 }
 
+// WithFields clones the client, keeping the underlying delegate the same but adding
+// fields to the logging context
+func (jc *client) WithFields(fields logrus.Fields) Client {
+	return &client{
+		transport: jc.transport,
+		upstream:  jc.upstream,
+		logger:    jc.logger.WithFields(fields),
+		delegate:  jc.delegate,
+	}
+}
+
+// ForPlugin clones the client, keeping the underlying delegate the same but adding
+// a plugin identifier and log field
+func (jc *client) ForPlugin(plugin string) Client {
+	pluginLogger := jc.logger.WithField("plugin", plugin)
+	retryingClient := retryablehttp.NewClient()
+	usedFlagTransport := &clientUsedTransport{
+		upstream: retryingClient.HTTPClient.Transport,
+	}
+	retryingClient.HTTPClient.Transport = usedFlagTransport
+	retryingClient.Logger = &retryableHTTPLogrusWrapper{log: pluginLogger}
+	// ignore error as url.String() was passed to the delegate
+	jiraClient, _ := newJiraClient(jc.url, jc.options, retryingClient)
```

Comment:
Can we log the error here? We don't expect it to be possible that this errors, but for example if the jira lib gets upgraded and then errors for whatever other reason, this could happen and would be a PITA to debug